--- a/config.mk
+++ b/config.mk
@@ -118,9 +118,10 @@ CONFIG_MAC80211=m
 # CONFIG_MAC80211_DRIVER_API_TRACER=y
 
 # Select in the next two lines which rate control algorithm to use
-CONFIG_MAC80211_RC_DEFAULT=cogtra
-CONFIG_COMPAT_MAC80211_RC_DEFAULT=cogtra
-CONFIG_MAC80211_RC_COGTRA=y
+CONFIG_MAC80211_RC_DEFAULT=cogtra_ht
+CONFIG_COMPAT_MAC80211_RC_DEFAULT=cogtra_ht
+CONFIG_MAC80211_RC_COGTRA=n
+CONFIG_MAC80211_RC_COGTRA_HT=y
 CONFIG_MAC80211_RC_CORA=n
 CONFIG_MAC80211_RC_MINSTREL=n
 CONFIG_MAC80211_RC_MINSTREL_HT=n
--- a/net/mac80211/Makefile
+++ b/net/mac80211/Makefile
@@ -54,6 +54,9 @@ rc80211_minstrel-$(CONFIG_MAC80211_DEBUG
 rc80211_minstrel_ht-y := rc80211_minstrel_ht.o
 rc80211_minstrel_ht-$(CONFIG_MAC80211_DEBUGFS) += rc80211_minstrel_ht_debugfs.o
 
+rc80211_cogtra_ht-y := rc80211_cogtra_ht.o
+rc80211_cogtra_ht-$(CONFIG_MAC80211_DEBUGFS) += rc80211_cogtra_ht_debugfs.o
+
 rc80211_cogtra-y := rc80211_cogtra.o
 rc80211_cogtra-$(CONFIG_MAC80211_DEBUGFS) += rc80211_cogtra_debugfs.o
 
@@ -69,6 +72,7 @@ rc80211_aarf-$(CONFIG_MAC80211_DEBUGFS)
 mac80211-$(CONFIG_MAC80211_RC_PID) += $(rc80211_pid-y)
 mac80211-$(CONFIG_MAC80211_RC_MINSTREL) += $(rc80211_minstrel-y)
 mac80211-$(CONFIG_MAC80211_RC_MINSTREL_HT) += $(rc80211_minstrel_ht-y)
+mac80211-$(CONFIG_MAC80211_RC_COGTRA_HT) += $(rc80211_cogtra_ht-y)
 mac80211-$(CONFIG_MAC80211_RC_COGTRA) += $(rc80211_cogtra-y)
 mac80211-$(CONFIG_MAC80211_RC_CORA) += $(rc80211_cora-y)
 mac80211-$(CONFIG_MAC80211_RC_ARF) += $(rc80211_arf-y)
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -1110,6 +1110,10 @@ static int __init ieee80211_init(void)
 	BUILD_BUG_ON(offsetof(struct ieee80211_tx_info, driver_data) +
 		     IEEE80211_TX_INFO_DRIVER_DATA_SIZE > sizeof(skb->cb));
 
+	ret = rc80211_cogtra_ht_init();
+	if (ret)
+		return ret;	
+	
 	ret = rc80211_cogtra_init();
 	if (ret)
 		return ret;
@@ -1157,6 +1161,8 @@ static int __init ieee80211_init(void)
 	rc80211_cora_exit();
  err_cora:
 	rc80211_cogtra_exit();
+ err_cogtra:
+	rc80211_cogtra_ht_exit();
 
 	return ret;
 }
@@ -1166,6 +1172,7 @@ static void __exit ieee80211_exit(void)
 	rc80211_pid_exit();
 	rc80211_minstrel_ht_exit();
 	rc80211_minstrel_exit();
+	rc80211_cogtra_ht_exit();
 	rc80211_cogtra_exit();
 	rc80211_cora_exit();
 	rc80211_arf_exit();
--- a/net/mac80211/rate.h
+++ b/net/mac80211/rate.h
@@ -186,6 +186,19 @@ static inline void rc80211_cogtra_exit(v
 }
 #endif
 
+#ifdef CONFIG_MAC80211_RC_COGTRA_HT
+extern int rc80211_cogtra_ht_init(void);
+extern void rc80211_cogtra_ht_exit(void);
+#else
+static inline int rc80211_cogtra_ht_init(void)
+{
+	return 0;
+}
+static inline void rc80211_cogtra_ht_exit(void)
+{
+}
+#endif
+
 #ifdef CONFIG_MAC80211_RC_ARF
 extern int rc80211_arf_init(void);
 extern void rc80211_arf_exit(void);
